File: BJIT_all_Projects/30061-Sajal_EMS/YSD_B02_J2EE_FinalProject_Sajal/frontend/lib/models/weightage.dart

import 'dart:convert';

Weightage weightageFromJson(String str) => Weightage.fromJson(json.decode(str));

String weightageToJson(Weightage data) => json.encode(data.toJson());

class Weightage {
  int? id;
  double? dailyTaskEvaluationWeightage;
  double? miniProjectEvaluationWeightage;
  double? midProjectEvaluationWeightage;
  double? finalProjectEvaluationWeightage;
  double? domainWeightage;
  double? managerEvaluationWeightage;
  double? trainingWeightage;
  double? hrInterviewEvaluationWeightage;

  Weightage({
    this.id,
    this.dailyTaskEvaluationWeightage,
    this.miniProjectEvaluationWeightage,
    this.midProjectEvaluationWeightage,
    this.finalProjectEvaluationWeightage,
    this.domainWeightage,
    this.managerEvaluationWeightage,
    this.trainingWeightage,
    this.hrInterviewEvaluationWeightage,
  });

  factory Weightage.fromJson(Map<String, dynamic> json) => Weightage(
    id: json["id"],
    dailyTaskEvaluationWeightage: json["dailyTaskEvaluationWeightage"],
    miniProjectEvaluationWeightage: json["miniProjectEvaluationWeightage"],
    midProjectEvaluationWeightage: json["midProjectEvaluationWeightage"],
    finalProjectEvaluationWeightage: json["finalProjectEvaluationWeightage"],
    domainWeightage: json["domainWeightage"],
    managerEvaluationWeightage: json["managerEvaluationWeightage"],
    trainingWeightage: json["trainingWeightage"],
    hrInterviewEvaluationWeightage: json["hrInterviewEvaluationWeightage"],
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "dailyTaskEvaluationWeightage": dailyTaskEvaluationWeightage,
    "miniProjectEvaluationWeightage": miniProjectEvaluationWeightage,
    "midProjectEvaluationWeightage": midProjectEvaluationWeightage,
    "finalProjectEvaluationWeightage": finalProjectEvaluationWeightage,
    "domainWeightage": domainWeightage,
    "managerEvaluationWeightage": managerEvaluationWeightage,
    "trainingWeightage": trainingWeightage,
    "hrInterviewEvaluationWeightage": hrInterviewEvaluationWeightage,
  };
}

File: BJIT_all_Projects/30061-Sajal_EMS/YSD_B02_J2EE_FinalProject_Sajal/backend/src/main/java/com/bjitacademy/sajal48/ems/domian/evaluation/Weightage.java
package com.bjitacademy.sajal48.ems.domian.evaluation;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Weightage implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequence_generator")
    private Long id;
    private Double dailyTaskEvaluationWeightage;
    private Double miniProjectEvaluationWeightage;
    private Double midProjectEvaluationWeightage;
    private Double finalProjectEvaluationWeightage;
    private Double domainWeightage;
    private Double managerEvaluationWeightage;
    private Double trainingWeightage;
    private Double hrInterviewEvaluationWeightage;
}